examples/103_transformer_layout_options.ipynb

```diff
@@ -0,0 +1,252 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hierarchical Diagram with Ports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hier_tree = networkx.readwrite.json_graph.node_link_graph(\n",
+    "    json.loads(pathlib.Path(\"hier_tree.json\").read_text(encoding=\"utf-8\"))\n",
+    ")\n",
+    "hier_ports = networkx.readwrite.json_graph.node_link_graph(\n",
+    "    json.loads(pathlib.Path(\"hier_ports.json\").read_text(encoding=\"utf-8\"))\n",
+    ")\n",
+    "\n",
+    "hier_elk = ipyelk.ElkDiagram()\n",
+    "hier_xelk_layout = ipyelk.nx.XELKTypedLayout()\n",
+    "hier_xelk = ipyelk.nx.XELK(\n",
+    "    source=(hier_ports, hier_tree), layouts=hier_xelk_layout.value\n",
+    ")\n",
+    "\n",
+    "\n",
+    "def _type_opt_change(change):\n",
+    "    hier_xelk.layouts = hier_xelk_layout.value\n",
+    "    hier_xelk.refresh()\n",
+    "\n",
+    "\n",
+    "hier_xelk_layout.observe(_type_opt_change, \"value\")\n",
+    "hier_xelk.connect(hier_elk)\n",
+    "\n",
+    "\n",
+    "toggle = ipywidgets.Button(description=\"Toggle Collapsed\")\n",
+    "\n",
+    "\n",
+    "@toggle.on_click\n",
+    "def toggle_node(widget):\n",
+    "    for element_id in hier_elk.selected:\n",
+    "        if element_id in hier_tree:\n",
+    "            for child in hier_tree.neighbors(element_id):\n",
+    "                state = hier_tree.nodes[child].get(\"hidden\", False)\n",
+    "                hier_tree.nodes[child][\"hidden\"] = not state\n",
+    "            hier_xelk.refresh()\n",
+    "\n",
+    "\n",
+    "ipywidgets.VBox(\n",
+    "    [\n",
+    "        ipywidgets.HBox(\n",
+    "            [ipywidgets.HTML(\"<h2>👇 click a group node then click 👉</h2>\"), toggle]\n",
+    "        ),\n",
+    "        hier_elk,\n",
+    "        hier_xelk_layout,\n",
+    "    ]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "JSON(hier_xelk_layout.value)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hier_xelk_layout.children = hier_xelk_layout._ui()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "JSON(hier_elk.value)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 🦌 Learn More 📖\n",
+    "\n",
+    "- [🦌 Introducing ELK 👋](./00_Introduction.ipynb)\n",
+    "- [🦌 Linking ELK Diagrams 🔗](./01_Linking.ipynb)\n",
+    "- [🦌 ELK Transformer 🤖](./02_Transformer.ipynb)\n",
+    "- [🦌 ELK App 🚀](./03_App.ipynb)\n",
+    "- [🦌 Interactive ELK App 🕹️](./04_Interactive.ipynb)\n",
+    "- [🦌 Node Label Placement 🏷️](./100_node_label_placement.ipynb)\n",
+    "- [🦌 Text Sizer 📏](./101_text_sizer.ipynb)"
```

Comment:
re #28: we could have a table of contents that we just import instead of keeping this up-to-date